### PR TITLE
Add some more dartdocs

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter/lib/src/animation/animation_controller.dart [hunk 1]
```diff
@@ -164,6 +164,7 @@ class AnimationController extends Animation<double>
   @override
   double get value => _value;
   double _value;
+
```
Comment (Hixie) [inline packages/flutter/lib/src/animation/animation_controller.dart]: why the newline? it's part of the value "block"...

(i'm fine with it either way, but that's why it didn't have a newline before)

File: packages/flutter/lib/src/material/app_bar.dart [hunk 2]
```diff
@@ -255,8 +255,14 @@ class AppBar extends StatefulWidget {
   /// Defaults to being adapted to the current [TargetPlatform].
   final bool centerTitle;
 
+  /// How opaque the toolbar part of the app bar is.
+  ///
+  /// A value of 1.0 is fully opaque, and a value of 0.0 is fully transparent.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/app_bar.dart]:
```md
Typically this is not changed from its default value (1.0). It is used by [SliverAppBar]
to animate the opacity of the toolbar when the app bar is scrolled.
```

Similarly for bottomOpacity below.

File: packages/flutter/lib/src/material/tabs.dart [hunk 3]
```diff
@@ -312,14 +312,27 @@ class _ChangeAnimation extends Animation<double> with AnimationWithParentMixin<d
...
 ///
-/// If a [TabController] is not provided, then there must be a [DefaultTabController]
-/// ancestor.
+/// Typically created as part of an [AppBar] and in conjuction with a
+/// [TabBarView].
+///
+/// If a [TabController] is not provided, then there must be a
+/// [DefaultTabController] ancestor.
 ///
 /// Requires one of its ancestors to be a [Material] widget
```
Comment (Hixie) [inline packages/flutter/lib/src/material/tabs.dart]: missing period on this line

File: packages/flutter/lib/src/painting/text_painter.dart [hunk 4]
```diff
@@ -127,6 +128,8 @@ class TextPainter {
   }
 
   ui.Paragraph _layoutTemplate;
+
+  /// The height of a zero-width space in [style].
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/text_painter.dart]: Units?
Maybe also include motivation for exposing this.

File: packages/flutter/lib/src/rendering/error.dart [hunk 5]
```diff
@@ -9,8 +9,9 @@ import 'object.dart';
 
 const double _kMaxWidth = 100000.0;
 const double _kMaxHeight = 100000.0;
+
 // Line length to fit small phones without dynamically checking size.
-const String line = '\n────────────────────\n\n';
+const String _kLine = '\n────────────────────\n\n';
```
Comment (Hixie) [inline packages/flutter/lib/src/rendering/error.dart]: while you're here we should probably also add another `\n` at the start there so it's more balanced.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/animation/animation_controller.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/animation/animation_controller.dart]: That makes sense.  I'll put it back.

**Pull request merged**